flax/nn/linear.py

```diff
@@ -219,12 +227,16 @@ def apply(self,
     """
 
     inputs = jnp.asarray(inputs, dtype)
+    if isinstance(kernel_size, int):
+      kernel_size = (kernel_size,)
```

Comment:
In this code path, users could also pass `(kernel_size, )` as the argument. Is that desirable? (I would imagine not, unless there's a strong reason)

I'd prefer to see the argument be an int OR a tuple of the right length.

Reply:
The standard way of allowing a int as a short hand for (int,) in the numpy, jax, and flax api is to allow both `i` and `(i,)`. Sometimes code creates tuples of dynamic size so having to "unpack" a size 1 tuple into an int would be annoying for users.

Reply:
TIL something :)